site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/service/SiteServiceImpl.java

```diff
@@ -42,32 +43,44 @@ public SiteServiceImpl(EntityStore<Site> siteStore,
         }
         this.siteStore = siteStore;
         this.entityValidation = entityValidation;
-
         logger.info(Services.STARTUP.message());
-
     }
 
     /**
-     *
      * @param site the Site to save.
      * @return The id of the new site
      */
-    public String save(Site site) {
+    @Override
+    public String save(final Site site) {
         var validationResponse = entityValidation.validate(site);
         if (!validationResponse.isValid()) {
             throw new ValidationException(validationResponse.getErrorMessage());
         }
+        if (findSiteByName(site.getName()).isPresent()) {
+            throw new ValidationException(Services.SITE_EXISTS.message());
+        }
+        if (site.getParentSiteId() == null) {
+            if (isRootSitePresent()) {
+                throw new ValidationException(Services.ONE_ROOT_SITE.message());
+            }
+        } else {
```

Comment:
This can be simplified with a single if for both the null check and the present check.
And, since it throws an exception, you can delete the else.

Reply:
I don't think so, because the else contains the validation on parentSiteId, which we don't want to run if it is null (regardless of whether the root is present)? So logic is: if (no parent site ID specified) then (throw if already a root site) else (validate parent site ID)